pkg/sources/s3/s3.go

```diff
@@ -230,10 +275,33 @@ func (s *Source) scanBuckets(
 
 		errorCount := sync.Map{}
 
+		input := &s3.ListObjectsV2Input{Bucket: &bucket}
+		if bucket == resumePoint.CurrentBucket && resumePoint.StartAfter != "" {
+			input.StartAfter = &resumePoint.StartAfter
+			ctx.Logger().V(3).Info(
+				"Resuming bucket scan",
+				"start_after", resumePoint.StartAfter,
+			)
+		}
```

Comment:
I thought this was handled above, but I guess it's not: What happens if the bucket at the resume point was ignored and the scan was restarted? I.e.
1. Bucket `x` is being scanned
2. The scan is terminated
3. The config is modified to ignore bucket `x`
4. The scan resumes

Reply:
Updated the logic. Let me know if that's clear.